recipe/meta.yaml

```diff
@@ -11,48 +11,43 @@ source:
   sha256: d1877192b44d36055f39fe212ca0abf04416bac1ff81be1b6cda0fbe82c12ac5
 
 build:
-  number: 1
+  number: 2
   noarch: python
   script: {{ PYTHON }} -m pip install . -vv
 
 requirements:
   host:
     - pip
-    - python >=3.6.2
+    - python >=3.7
   run:
-    - addict >=2.4.0
-    - jsonschema ==3.2.0
+    - jsonschema >=3.2.0
     - jstyleson >=0.0.2
     - natsort >=7.1.0
-    - networkx >=2.5
-    - ninja >=1.10.0.post2
-    - numpy >1.19.2
+    - networkx >=2.6, <=2.8.2
+    - ninja >=1.10.0.post2, <1.11
+    - numpy >1.19.2, <1.25
+    - openvino-telemetry >=2023.1.0
+    - packaging >=20.0
+    - pandas >=1.1.5, <2.1
+    - pip
+    - pymoo >=0.6.0
+    - psutil
     - pydot >=1.4.1
-    - python >=3.6.2
+    - pyparsing <3.0
+    - python >=3.7
     - scikit-learn >=0.24.0
-    - scipy >=1.3.2
-    # - scipy >=1.3.2  # [py>=37]
-    # - scipy >=1.3.2,<=1.5.4  # [py<37]
-    - matplotlib-base
-    # - matplotlib-base >=3.3.4  # [py>=37]
-    # - matplotlib-base >=3.3.4,<3.4  # [py<37]
-    - openvino-telemetry >=2023.1.1
-    - pandas
-    # - pandas >=1.1.5  # [py>=37]
-    # - pandas >=1.1.5,<1.2  # [py<37]
+    - scipy >=1.3.2, <1.11
     - texttable >=1.6.3
-    # - pytorch >=1.5.0,<=1.8.1,!=1.8.0
```

Comment:
Since around v2.x we have backend dependencies as runtime-dependencies, if no supported backends (e.g. torch, tensorflow, onnx etc.) are found, we spawn an error message to the log:
```
import: 'nncf'
ERROR:nncf:Neither PyTorch, TensorFlow, ONNX or OpenVINO Python packages have been found in your Python environment.
Please install one of the supported frameworks above in order to use NNCF on top of it.
See the installation guide at https://github.com/openvinotoolkit/nncf#installation for help.
```
but otherwise the installation itself can be considered a success.

We specify exact BKC backend versions in setup.py via the `extras` syntax. Couldn't find out yet how to do the same in a conda recipe, might have to do this in a follow-up PR.